query-engine/driver-adapters/js/smoke-test-js/src/client/client.ts

```diff
@@ -48,34 +47,25 @@ export async function smokeTestClient(driverAdapter: DriverAdapter) {
           '-- Implicit "COMMIT" query via underlying driver',
         ]
 
-        const postgresExpectedQueries = [
-          'BEGIN',
-          'DEALLOCATE ALL',
-          'SELECT 1',
-          'SELECT 2',
-          'SELECT 3',
-          'COMMIT',
-        ]
-
         if (['mysql'].includes(provider)) {
           if (isUsingDriverAdapters) {
             assert.deepEqual(queries, driverAdapterExpectedQueries)
           } else {
             assert.deepEqual(queries, defaultExpectedQueries)
           }
         } else if (['postgres'].includes(provider)) {
-          if (isUsingDriverAdapters) {
-            assert.deepEqual(queries, defaultExpectedQueries)
-          } else {
-            assert.deepEqual(queries, postgresExpectedQueries)
-          }
+          // Note: the "DEALLOCATE ALL" query is only present after "BEGIN" when using Rust Postgres with pgbouncer.
+          assert.deepEqual(queries.at(0), defaultExpectedQueries.at(0))
```

Comment:
probably not necessary anymore after rewriting the second assertion but I'll let you decide 😄

Reply:
This is actually useful to ensure that `DEALLOCATE ALL` appears after `BEGIN` (if at all), which could change if we introduced bugs in how e.g. the adapters' Postgres handles reset queries in `Queryable` + `TransactionCapable`.